src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -270,7 +270,7 @@ public static int LastIndexOf(ref byte searchSpace, int searchSpaceLength, ref b
                                 return bitPos + offset;
                             }
                             // Clear the highest set bit.
-                            mask = BitOperations.ResetBit(mask, bitPos);
+                            mask = BitOperations.ToggleBit(mask, bitPos);
```

Comment:
This is one less operation, but its also "less clear" and "more error prone" overall as refactorings or other things may end up breaking this accidentally.

I think if we did this, we should have an assert that the bit is actually cleared.

Reply:
And maybe name it ClearSetBit so it's clearer at the call site?

Reply:
> And maybe name it ClearSetBit so it's clearer at the call site?

If it was named `ClearSetBit` it should be private `SpanHelpers` method with an assert, rather than an internal `BitOperations` method.

But I think the best approach is to rename `bitPos` as `highestSetBitIndex`. This is also less ambiguous as to whether numbering of the position starts at 0 or 1.

Reply:
I think the latter is probably better since it lets us have more generally reusable code